contracts/test/integration/vip/vip13.sol

```diff
@@ -0,0 +1,477 @@
+    int8 private constant usdcDecimalNormalizer = 12;
+
+    /// @notice no scaling to do on DAI as decimals are 18
+    int8 private constant daiDecimalNormalizer = 0;
+
+    ERC20Allocator private allocator =
+        ERC20Allocator(MainnetAddresses.ERC20ALLOCATOR);
+
+    ERC20CompoundPCVDeposit private daiDeposit =
+        ERC20CompoundPCVDeposit(MainnetAddresses.COMPOUND_DAI_PCV_DEPOSIT);
+    ERC20CompoundPCVDeposit private feiDeposit =
+        ERC20CompoundPCVDeposit(MainnetAddresses.COMPOUND_FEI_PCV_DEPOSIT);
+    ERC20CompoundPCVDeposit private usdcDeposit =
+        ERC20CompoundPCVDeposit(MainnetAddresses.COMPOUND_USDC_PCV_DEPOSIT);
+
+    constructor() {}
+
+    function getMainnetProposal()
+        public
+        override
+        returns (ITimelockSimulation.action[] memory prop)
+    {
+        voltV2 = new VoltV2(MainnetAddresses.CORE);
+        voltMigrator = new VoltMigrator(MainnetAddresses.CORE, address(voltV2));
+        deployPsms(address(voltV2));
+
+        migratorRouter = new MigratorRouter(
+            address(voltV2),
+            address(voltMigrator),
+            voltV2DaiPriceBoundPSM,
+            voltV2UsdcPriceBoundPSM
+        );
+
+        address[] memory toWhitelist = new address[](3);
+        toWhitelist[0] = address(voltV2UsdcPriceBoundPSM);
+        toWhitelist[1] = address(voltV2DaiPriceBoundPSM);
+        toWhitelist[2] = address(voltMigrator);
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.CORE,
+                arguments: abi.encodeWithSignature(
+                    "grantMinter(address)",
+                    MainnetAddresses.TIMELOCK_CONTROLLER
+                ),
+                description: "Grant minter role to timelock"
+            })
+        );
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: address(voltV2),
+                arguments: abi.encodeWithSignature(
+                    "mint(address,uint256)",
+                    address(voltMigrator),
+                    voltInDaiPSM + voltInUsdcPSM
+                ),
+                description: "Mint new volt for new VOLT to migrator contract"
+            })
+        );
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.CORE,
+                arguments: abi.encodeWithSignature(
+                    "revokeMinter(address)",
+                    MainnetAddresses.TIMELOCK_CONTROLLER
+                ),
+                description: "Remove minter role from timelock"
+            })
+        );
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: address(MainnetAddresses.VOLT),
+                arguments: abi.encodeWithSignature(
+                    "approve(address,uint256)",
```

Comment:
Why do we need to approve the migrator to spend the timelock's volt if the this governance action just mints all Volt to the migrator?

Reply:
We need to approve the router so that the timelock can carry out the exchange using the migrator